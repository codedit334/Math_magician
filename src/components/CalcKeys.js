import React, { useEffect, useState } from 'react';
import CalcInput from './CalcInput';
import calculate from '../logic/calculate';
import '../styles/calcKeys.css';

export default function CalcKeys() {
  const [initialObj, setinitialObj] = useState({
    total: 0,
    next: 0,
    operation: null,
  });

  const calculatorKeys = [
    ['AC', '+/-', '%', 'รท'],
    ['7', '8', '9', 'x'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '='],
  ];

  let table = calculatorKeys.map((row) => {
    const tableData = row.map(
      (data) => `<td ${+data === 0 && 'colspan="2"'} >${data}</td>` // eslint-disable-line
    ); // eslint-disable-line

    const finalData = `<tr>${tableData}</tr>`;

    return finalData;
  });

  table = table.toString().replace(/,/g, '');

  useEffect(() => {
    document.querySelector('tbody').innerHTML = table;
    document.querySelectorAll('td').forEach((element) => {
      element.addEventListener('click', (e) => {
        const clickedBtn = e.target.innerText;
        const gotValue = calculate(initialObj, clickedBtn);
        setinitialObj(gotValue);
      });
    });
  }, [initialObj]);

  return (
    <>
      <CalcInput resObj={initialObj} />
      <div className="calcKeys">
        <table>
          <tbody />
        </table>
      </div>
    </>
  );
}
