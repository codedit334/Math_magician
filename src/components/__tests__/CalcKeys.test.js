import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalcKeys from '../CalcKeys';

test('Sould render CalcKeys Component', () => {
  render(<CalcKeys />);
  const CalcKeysTBodyElement = screen.getByTestId('CalcKeysTBody');
  expect(CalcKeysTBodyElement).toBeInTheDocument();
  expect(CalcKeysTBodyElement).toHaveTextContent('AC');

  const tree = renderer.create(<CalcKeys />).toJSON();
  expect(tree).toMatchSnapshot();
});
