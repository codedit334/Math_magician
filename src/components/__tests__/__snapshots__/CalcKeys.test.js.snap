// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sould render CalcKeys Component 1`] = `
Array [
  <div
    className="calcinput"
  >
    <input
      className="calcinput-display"
      data-testid="CalcInput"
      disabled={true}
      placeholder={0}
      value=""
    />
  </div>,
  <div
    className="calcKeys"
  >
    <table>
      <tbody
        data-testid="CalcKeysTBody"
      />
    </table>
  </div>,
]
`;
